live-ingester/src/test/java/org/alfresco/hxi_connector/live_ingester/adapters/storage/connector/HttpFileUploaderIntegrationTest.java

```diff
@@ -51,59 +52,59 @@
-class StorageClientIntegrationTest
+class HttpFileUploaderIntegrationTest
 {
     private static final String LOCALSTACK_IMAGE = "localstack/localstack";
     private static final String LOCALSTACK_TAG = DockerTags.getOrDefault("localstack.tag", "3.0.2");
     private static final String BUCKET_NAME = "test-hxinsight-bucket";
     private static final String OBJECT_KEY = "dummy.txt";
-    private static final String OBJECT_CONTENT = "dummy's file dummy content";
+    private static final String OBJECT_CONTENT = "Dummy's file dummy content";
     private static final String OBJECT_CONTENT_TYPE = "plain/text";
 
     @Container
     @SuppressWarnings("PMD.FieldNamingConventions")
-    private static final LocalStackContainer localStack = new LocalStackContainer(DockerImageName.parse(LOCALSTACK_IMAGE).withTag(LOCALSTACK_TAG));
+    static final LocalStackContainer localStackServer = new LocalStackContainer(DockerImageName.parse(LOCALSTACK_IMAGE).withTag(LOCALSTACK_TAG));
 
     @Autowired
-    private LocalStorageClient testS3Storage;
+    LocalStorageClient s3StorageMock;
     @Autowired
-    private StorageClient storageClient;
+    FileUploader fileUploader;
 
     @BeforeAll
     static void beforeAll() throws IOException, InterruptedException
     {
-        localStack.execInContainer("awslocal", "s3api", "create-bucket", "--bucket", BUCKET_NAME);
+        localStackServer.execInContainer("awslocal", "s3api", "create-bucket", "--bucket", BUCKET_NAME);
     }
 
     @Test
     void testUpload() throws IOException
     {
-        List<String> freshBucketContent = testS3Storage.listBucketContent(BUCKET_NAME);
-        assertThat(freshBucketContent).isEmpty();
-
         // given
+        List<String> initialBucketContent = s3StorageMock.listBucketContent(BUCKET_NAME);
         @Cleanup
         InputStream fileInputStream = new ByteArrayInputStream(OBJECT_CONTENT.getBytes());
-        URL preSignedUrl = testS3Storage.generatePreSignedUploadUrl(BUCKET_NAME, OBJECT_KEY, OBJECT_CONTENT_TYPE);
+        URL preSignedUrl = s3StorageMock.generatePreSignedUploadUrl(BUCKET_NAME, OBJECT_KEY, OBJECT_CONTENT_TYPE);
+        FileUploadRequest fileUploadRequest = new FileUploadRequest(fileInputStream, OBJECT_CONTENT_TYPE, preSignedUrl);
 
         // when
-        int actualStatusCode = storageClient.upload(fileInputStream, OBJECT_CONTENT_TYPE, preSignedUrl).getStatusCode();
+        fileUploader.upload(fileUploadRequest);
 
         // then
-        assertThat(actualStatusCode).isEqualTo(200);
-        List<String> actualBucketContent = testS3Storage.listBucketContent(BUCKET_NAME);
-        assertThat(actualBucketContent).containsOnly(OBJECT_KEY);
+        List<String> actualBucketContent = s3StorageMock.listBucketContent(BUCKET_NAME);
+        assertThat(initialBucketContent).doesNotContain(OBJECT_KEY);
+        assertThat(actualBucketContent).contains(OBJECT_KEY);
+        assertThat(actualBucketContent.size() - initialBucketContent.size()).isEqualTo(1);
```

Comment:
Yes I did - it should preserve the test from failing in case of adding other test methods operating on the same S3 bucket. I've updated the test a bit and it's a bit neater now.